lib/rules/lint-table-groups.js

```diff
@@ -49,31 +45,22 @@ function isAllowedTableChild(node) {
         return ALLOWED_TABLE_CHILDREN.includes(tagNameAttribute.value.chars);
       }
       break;
+    case 'CommentStatement':
+    case 'MustacheCommentStatement':
+      return true;
     case 'TextNode':
       return !/\S/.test(node.chars);
   }
 
   return false;
 }
 
-function hasChildTags(tableNode) {
```

Comment:
This special check is what caused it to fail for comments next to valid children. It seems cleaner to remove this preemptive check entirely, so we don't have to keep two different checks consistent.